### PR TITLE
[OUTREACHY] Mobile UI optimization

### PR DESCRIPTION
**Pull request by kemsguy7**
This pr resolves issue: #5318  where i resolved some ux issues on the mobile view of the registry page. 

### **Key changes** 

- ***
- Changed display of filters and search input fields to flex column display for better visual alignment
- Fixed overflowing pill by taking it to a new line and reducing the font size

### **Demo**

https://github.com/user-attachments/assets/5ec7bff9-2fbf-4425-a38a-fff13bb63de0

### CODE REVIEW

**Comment by kemsguy7** [comment]
@svrnm , please kindly review

File: CONTRIBUTING.md
Comment (svrnm) [inline CONTRIBUTING.md:1]: please remove this unrelated change

File: assets/scss/_external_link.scss
Comment (svrnm) [inline assets/scss/_external_link.scss:1]: please remove this unrelated change

File: assets/scss/_styles_project.scss
Comment (svrnm) [inline assets/scss/_styles_project.scss:1]: please remove this unrelated change

File: assets/scss/_variables_project.scss
Comment (svrnm) [inline assets/scss/_variables_project.scss:1]: please remove this unrelated change

File: content/en/_index.md
Comment (svrnm) [inline content/en/_index.md:1]: please remove this unrelated change

File: content/en/ecosystem/registry/_index.md
Comment (svrnm) [inline content/en/ecosystem/registry/_index.md:1]: please remove this unrelated change

File: data/registry/resource-detector-go-aws.yml
Comment (svrnm) [inline data/registry/resource-detector-go-aws.yml:1]: please remove this unrelated change

File: layouts/shortcodes/ecosystem/registry/search-form.html [hunk 1]
```diff
@@ -43,14 +43,14 @@
...
-      <input class="form-control w-auto" list="search-suggestions" id="input-s" type="text" name="s" placeholder="Type to search…"
+      <span class="input-group-text mb-3" id="basic-addon1" title="{{ len $registry }} entries">Search {{ len $registry }} entries</span>
+      <input class="form-control mb-3 w-auto" list="search-suggestions" id="input-s" type="text" name="s" placeholder="Type to search…"
         aria-label="Registry search input field">
         <datalist id="search-suggestions"></datalist>
 
 
       <input type="hidden" name="component" id="input-component" value="" />
-      <input type="hidden" name="language" id="input-language" value="" />
+      <input type="hidden" name="language" id="input-language" value="" />
```
Comment (svrnm) [inline layouts/shortcodes/ecosystem/registry/search-form.html:53]: please remove this unrelated change

**Review by svrnm — CHANGES_REQUESTED** [review]
There are a lot of changes that are unrelated to your PR, please revert them.
